tools/mock-service-host/src/mid/coordinator.ts

```diff
@@ -282,17 +295,110 @@ export class Coordinator {
                 throw new HasChildResource(req.url)
             }
         } else {
-            const [code, _ret] = this.findResponse(exampleResponses, HttpStatusCode.OK)
+            let code, ret
+            if (lroCallback !== null || req.query?.[LRO_CALLBACK] === 'true') {
+                // for the lro operaion
+                if (req.query?.[LRO_CALLBACK] === 'true') {
+                    // lro callback, try to get 204 response first
+                    try {
+                        // if 201 response exist, need to return 200 response
+                        this.findResponse(exampleResponses, HttpStatusCode.CREATED)
```

Comment:
will be good to refactor this block of code to avoid much use too much of try..catch